service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt

```diff
@@ -464,6 +468,40 @@ class ChildAttendanceController(
         Audit.ChildAttendancesFullDayAbsenceCreate.log(targetId = childId, objectId = unitId)
     }
 
+    @DeleteMapping("/units/{unitId}/children/{childId}/full-day-absence")
+    fun cancelFullDayAbsence(
+        db: Database,
+        user: AuthenticatedUser,
+        clock: EvakaClock,
+        @PathVariable unitId: DaycareId,
+        @PathVariable childId: ChildId
+    ) {
+        val today = clock.today()
+
+        db.connect { dbc ->
+            dbc.transaction { tx ->
+                accessControl.requirePermissionFor(
+                    tx,
+                    user,
+                    clock,
+                    Action.Child.DELETE_ABSENCE,
+                    childId
+                )
+                val placementType =
+                    tx.fetchChildPlacementBasics(childId, unitId, clock.today()).placementType
+                val absenceCategories =
+                    tx.getAbsencesOfChildByDate(childId, today).map { it.category }.toSet()
+                val hasFullDayAbsence = placementType.absenceCategories() == absenceCategories
+                if (hasFullDayAbsence) {
+                    tx.deleteAbsencesByDate(childId, today)
+                } else {
+                    throw Conflict("Cannot cancel full day absence, child is not fully absent")
+                }
+            }
+        }
+        Audit.ChildAttendancesFullDayAbsenceDelete.log(targetId = childId, objectId = unitId)
```

Comment:
This audit log is not very useful right now, and should probably include ids of the deleted absence